rep-2000.rst

```diff
@@ -1,20 +1,69 @@
 REP: 2000
-Title: ROS 2 Target Platforms
-Author: Mikael Arguedas <mikael@openrobotics.org>, Steven! Ragnarok <steven@openrobotics.org>
+Title: ROS 2 Releases and Target Platforms
+Author: Mikael Arguedas <mikael@openrobotics.org>, Steven! Ragnarok <steven@openrobotics.org>, Dirk Thomas <dthomas@openrobotics.org>
 Status: Active
 Type: Informational
 Content-Type: text/x-rst
-Created: 10-Apr-2018, 24-Apr-2018, 21-May-2018, 10-Dec-2018, 20-May-2019, 21-May-2019, 29-Aug-2019
+Created: 10-Apr-2018, 24-Apr-2018, 21-May-2018, 10-Dec-2018, 20-May-2019, 21-May-2019, 29-Aug-2019, 25-Feb-2020
 
 
 Abstract
 ========
 
-This REP defines target platforms for each ROS 2 Distribution Release.
+This REP defines the timeline for future ROS 2 releases as well as the targeted platforms for each specific one.
 We define platforms to include both operating system releases (e.g. Ubuntu Xenial (16.04 LTS)) as well as major language releases (e.g. Python 3.5).
 The target platforms represent the set on which all core stacks are expected to work.
 Exceptions can be made for stacks that are intentionally platform-specific.
 
+Release Schedule
+================
+
+Note: the following applies to ROS 2 releases *after* Foxy Fitzroy.
+Before, releases were made more frequently but with shorter support due to the fact that many foundational parts of ROS 2 were still heavily iterated on.
+
+TL;DR ROS 2 release schedule post Foxy:
+* May 2021: G Turtle: non-LTS release, supported for 1.5 years
+* May 2022: H Turtle: LTS release, supported for 5 years
+* May 2023: I Turtle: non-LTS release, supported for 1.5 years
+* May 2024: J Turtle: LTS release, supported for 5 years
+* and so on, alternating annually between LTS and non-LTS releases
+
+Frequency
+---------
+
+New ROS 2 releases are published in a time based fashion **every 12 months**.
+The rationale is that a shorter cycle (like 6 months) results in a significant overhead and many active releases at the same time (assuming they have the same support length).
+On the other hand a longer cycle (like 2 years) is too long for users to wait for new features to be available in a ROS 2 release.
+
+Targeted Platforms
+------------------
+
+Since regular Ubuntu releases are only supported for 9 months ROS 2 is not targeting those.
```

Comment:
Has it been considered to support the latest non-LTS ubuntu on non-LTS ROS distributions ?

From a maintainer's perspective, non-LTS ROS distros are a good place to change API and try new things while keeping a deprecation cycle. It is also an opportunity to start supporting more recent dependencies and exploring their featureset.
With the current approach, maintainers will have to do a 2-year jump for system dependencies even if they target the non-LTS ROS distribution (and even the rolling release IIUC) and will have to do so in a short time period.

Could that impact the development of new features?
(maybe it applied only to me) If I have only a couple months to update my libraries, I will most likely just adapt to new API but not explore and leverage new features.

Reply:
> Has it been considered to support the latest non-LTS ubuntu

Yes, but Ubuntu non-LTS distros have a too short support length and are being used by too few users to justify the necessary effort.